File: Problems/0001_0099/0069_MySqrt/Project_CS/MySqrt.cs
using System;

public class Solution
{
    public int MySqrt(int x)
    {
        // 36ms
        if (x == 0)
            return 0;
        int left = 1, right = x;
        int ans = 0;
        while (left <= right) {
            int mid = left + (right - left) / 2;
            if (mid <= x / mid) {
                left = mid + 1;
                ans = mid;
            } else {
                right = mid - 1;
            }
        }
        return ans;
    }

    public int MySqrt_work(int x)
    {
        int i = 1;

        while ( (uint)i*i <= x)
            i += 1000;

        i -= 1000;
        while ( (uint)i*i <= x)
            i += 100;

        i -= 100;
        while ( (uint)i*i <= x)
            i += 10;

        i -= 10;
        while ( (uint)i*i <= x)
            i++;

        return (i - 1);
    }

    public void Main(string args)
    {
        int x  = int.Parse(args.Replace("[","").Replace("]","").Trim());

        System.Diagnostics.Stopwatch sw = new System.Diagnostics.Stopwatch();

        sw.Start();

        int result = MySqrt(x);
        Console.WriteLine("Result = " + result.ToString() );

        sw.Stop();

        Console.WriteLine("Execute time : " + sw.ElapsedMilliseconds.ToString() + "ms\n");
    }
}

File: Problems/0001_0099/0069_MySqrt/Project_Python3/MySqrt.py
# coding: utf-8

import math
import os
import sys
import time

class Solution:
#   def mySqrt(self, x: int) -> int:
    def mySqrt(self, x):
        # 32ms
        if x == 0:
            return 0
        left, right = 1, x
        while left <= right:
            mid = left + (right - left) // 2
            if mid <= x / mid:
                left = mid + 1
                ans = mid
            else:
                right = mid - 1
        return ans

    def mySqrt0(self, x):
        # 28ms
        if x < 2:
            return x
        else:
            return int(math.sqrt(x))

def main():
    argv = sys.argv
    argc = len(argv)

    if argc < 2:
        print("Usage: python {0} <testdata.txt>".format(argv[0]))
        exit(0)

    if not os.path.exists(argv[1]):
        print("{0} not found...".format(argv[1]))
        exit(0)

    testDataFile = open(argv[1], "r")
    lines = testDataFile.readlines()

    for temp in lines:
        temp = temp.strip()
        if temp == "":
            continue
        print("args = {0}".format(temp))
        loop_main(temp)
    #   print("Hit Return to continue...")
    #   input()

def loop_main(temp):
    fld = temp.replace("[","").replace("]","").replace("\"","").replace(" ","")

    n = int(fld)
    print("n = {0:d}".format(n))

    sl = Solution()
    time0 = time.time()

    result = sl.mySqrt(n)

    time1 = time.time()

    print("result = {0}".format(result))
    print("Execute time ... : {0:f}[s]\n".format(time1 - time0))

if __name__ == "__main__":
    main()
